apps/www/registry/default/ui/carousel.tsx

```diff
@@ -42,6 +49,47 @@ function useCarousel() {
   return context
 }
 
+export const useDotButton = (
+  emblaApi: CarouselApi | undefined
+): UseDotButtonType => {
+  const [selectedIndex, setSelectedIndex] = React.useState(0)
+  const [scrollSnaps, setScrollSnaps] = React.useState<number[]>([])
+
+  const onDotButtonClick = React.useCallback(
+    (index: number) => {
+      if (!emblaApi) return
+      emblaApi.scrollTo(index)
+    },
+    [emblaApi]
+  )
+
+  const onInit = React.useCallback((emblaApi: CarouselApi) => {
+    if (!emblaApi) return
+    setScrollSnaps(emblaApi.scrollSnapList())
+  }, [])
+
+  const onSelect = React.useCallback((emblaApi: CarouselApi) => {
+    if (!emblaApi) return
+    setSelectedIndex(emblaApi.selectedScrollSnap())
```

Comment:
With [`Autoplay`](https://www.embla-carousel.com/plugins/autoplay/) enabled, I had to add this line to reset its timer, which I believe is the user's expectation. They don't want to navigate only for the autoplay to immediately navigate them again.
```tsx
emblaApi.plugins().autoplay?.reset()
```